\begin{dissertationintroduction}
    As the defining structure of all chordates, the notochord plays a key role in signaling and coordinating development during embryogenesis. In most vertebrates, the notochord is replaced by ossified vertebrae, but it persists throughout the life of invertebrate chordates. My thesis focuses on understanding gene regulation in the notochord during embryonic development from both the perspective of the genomic sequence and the perspective of active proteins within this key structure. 
    
    \section{Notochord development in \textit{Ciona intestinalis}}
    Chordates are animals belonging to the phylum Chordata, which includes vertebrates (subphylum Vertebrata), tunicates (subphylum Tunicata), and cephalochordates (subphylum Cephalochordata) \cite{holland2005}. The key defining characteristic of all chordates is the presence of a notochord during embryonic development \cite{stemple2004, holland2005, stemple2005, corallo2015, balmer2016, debree2018}. The notochord is a long, semi-rigid fibrous rod of mesodermal origin that provides structural support to the developing embryo along the anterior-posterior (A–P) axis \cite{stemple2005, corallo2015, balmer2016}. It is surrounded by a sheath (the “notochordal” sheath), which is composed of extracellular matrix proteins \cite{corallo2015, stemple2004, stemple2005}

    The flexibility and rigidity of the notochord provide the basis for controlled mechanical support of Chordates and protection for the developing neural tube \cite{stemple2004, stemple2005, corallo2015, balmer2016}. While some Chordates retain the notochord throughout life as their body’s primary axial support, in most vertebrates, the notochord becomes the nucleus pulposus of the intervertebral disc \cite{stemple2005, corallo2015, lawson2015, balmer2016}. Failure of vertebral notochord cells to transition to the nucleus pulposus is associated with chordomas. These slow-growing tumors form from notochord cell remnants within the spine, or the base of the skull \cite{corallo2015, debree2018}. A complete understanding of notochord structure and function during early and late life stages is thus essential to better understand congenital vertebral defects.

    As a close chordate relative to the vertebrates, the ascidian Ciona intestinalis Type A or Ciona robusta (Ciona) stands as a longstanding model for studying organogenesis in a simple embryo \cite{dehal2002, delsuc2006, imai2006, satoh2014, satou2019, winkley2020}. For example, the Ciona notochord consists of only 40 post-mitotic cells, and orthologs of many Ciona notochord genes have known notochord expression in vertebrate embryos \cite{reeves2017, satoh2014, winkley2020}. Of the 40 notochord cells, 32 are grouped in the anterior of the body and compose the “primary” or “A-line” notochord. The remaining eight are located more posteriorly and form the “secondary” or “B-line” notochord \cite{satoh2014, winkley2020}. A-line and B-line refer to the conventional nomenclature denoting particular cell lineages in Ciona. 

    Within Ciona, notochord precursor cells are defined as early as the eight-cell stage as the A4.1 and B4.1 blastomere pair in the developing anterior and posterior regions of the embryo, respectively \cite{corbo1997, satoh2014, yagi2004}. The A4.1 cells then divide to form the A5.1 and A5.2 blastomere pair at the onset of the 16-cell stage, which are precursors to the A-line notochord and the endoderm, nerve cord, trunk lateral cells, and muscle \cite{satoh2014}. On the other hand, the B4.1 cells divide to form the B5.1 and B5.2 blastomere pair and, through subsequent divisions from B5.1, divide into B6.1 and B6.2. Finally, the B6.1 blastomere descendant at the 32-cell stage will eventually develop into the B-line notochord and other mesenchymal and muscle cells \cite{corbo1997, satoh2014, yagi2004}. When gastrulation initiates at the 110-cell stage, the Ciona embryo contains 16 primary and four secondary notochord precursor cells \cite{satoh2014}. Gastrulation is the stage at which the structure of the embryo changes from a single-layered blastula into a multiple-layered gastrula; thus, the notochord precursor cells coordinately invaginate as a monolayer over the primary gut, or archenteron \cite{rhee2005, winkley2020}. Following gastrulation is neurulation, the stage at which the embryonic neural plate develops and then forms the neural tube \cite{rhee2005, satoh2014}. At this stage, the notochord precursor cells in the Ciona embryo divide for the last time to define the final set of notochord cells on the embryonic midline \cite{nakamura2012}. 

    \section{Elucidating the mechanisms regulating notogenesis}
    The massive developmental transitions during embryogenesis require accurate gene regulation to maintain and balance the differentiation process. One component of this machinery is the interactions between cis-acting DNA elements–such as promoters and enhancers–and regulatory transcription factors. With the first example discovered in the early 1980s, enhancers are short regions of DNA that proteins can bind to regulate gene transcription \cite{khoury1983, kvon2021, levine2010}. Enhancers are typically located distally from the gene promoter and are approximately 100 bp to 1,000 bp in length \cite{khoury1983, levine2010}. Additionally, it has been suggested that depending on the syntax–the order, orientation, and spacing of transcription factor binding sites–and binding affinity, gene expression patterns can be finely controlled through a mechanism called “enhancer grammar” \cite{arnone1997, jindal2021}. Despite the importance of enhancers and their known association with developmental defects and disease, we still do not entirely understand how an enhancer’s sequence encodes particular functions. In Chapter 1, I will discuss the investigation into a notochord enhancer governed by a specific organization of Zic, ETS, FoxA, and Brachyury transcription factor binding sites \cite{farley2016, song2022}. The study which discovered this grammar found an interplay between the syntax and affinity of the Zic and ETS binding sites present, such that the organization could compensate for the affinity and vice versa \cite{farley2016}. Chapter 1 will discuss my continuation of this study in which I search for evidence of the Zic and ETS notochord enhancer grammar across the Ciona genome and test for functionality in the embryonic Ciona notochord through a pilot screen of 90 putative enhancer variants at the tailbud stage \cite{song2022}. 
    
    Beyond the universal quality of containing transcription factor motifs, enhancer sequences can vary significantly in the location, length, and type of transcription factor binding sites present. Additionally, these changes can be even more dramatic as you compare across species \cite{villar2015, ward2012, wong2020}. However, studies have suggested that even with low sequence conservation, the function of specific enhancers may be conserved across species and that this function may be partly due to combinatorial action of conserved transcription factors \cite{claussnitzer2014, wong2020}. This may be because a single transcription factor across its homologs in multiple species may have similar binding properties and thus recognize identical DNA sequences \cite{peter2011, wong2020}. In Chapter 2, I will continue the discussion of the notochord enhancer studied in Chapter 1 but in greater detail and at a larger scale across the Ciona genome. I evaluate the functionality of over four-thousand enhancer variants containing the notochord enhancer grammar of interest through a massively-parallel reporter assay. Chapter 2 offers a deeper investigation into the Zic, ETS, FoxA, and Brachyury organization and how its appearance in other vertebrate species may provide insight into the conservation of notochord regulation in chordates.

    Within Chapter 1 and Chapter 2, I discuss the importance of enhancer grammar as a mechanism in genome regulation. Nonetheless, understanding the underlying processes driving development also requires understanding how genes are expressed, primarily how these gene expression profiles differ across cells \cite{peter2011}. For instance, all cells in a developing embryo contain the same set of genes. However, different cells express different sets of these genes, leading to differences in expression and, thus, molecular function \cite{arnone1997, peter2011}. Technological advances have enabled the cataloging of global gene expression profiles of single cells using single-cell RNA-sequencing (scRNA-seq), allowing scientists to define the heterogeneity within cell populations during embryonic development \cite{klein2015a, macosko2015, olsen2018}. This new paradigm has allowed developmental biologists to identify precisely when and in which cell types controlling cell fate decisions are expressed \cite{klein2019}. Despite the availability of large cell-type atlases generated via scRNA-seq and other omics technologies, there is still much to be learned about gene regulatory networks. Ciona is a particularly suitable model for understanding the transcriptional changes necessary for proper development due to its genomic and morphological simplicity and historical significance as a model organism for embryological studies. In Chapter 3, I discuss an initiative to develop a high-resolution, single-cell atlas of a gastrulating Ciona embryo to understand notogenesis and the formation of other early structures. Additionally, I reflect on the potential of single-cell technologies to study genome regulation and enhancer grammar in a high-throughput and high-resolution manner. 
    
    \section{Training the next generation of bioinformaticians}
    In recent years, genomics technologies have become more high-throughput and affordable to all research groups, resulting in a boom in data available for all biomedical research areas. However, this also results in a backlog of data to analyze for those that conducted the experiments. Despite never receiving a formal education in computation, many researchers are then faced with the arduous task of learning how to run bioinformatics pipelines \cite{barone2017, stephens2015}. While computational courses have started being integrated into the standard curriculum for undergraduate Biology majors, there remains a need to support graduate students and other scientists that did not experience this shift in training for the field. In Chapter 4, I will discuss the pedagogical philosophy that drove the in-person, hybrid, and virtual Bioinformatics courses I taught at the University of California, San Diego. 

    \section{Conclusion}
    The massive developmental transitions during embryogenesis require accurate gene regulation to maintain and balance the differentiation process. In this dissertation, I present our approach to understanding regulation in the developing notochord by looking for genomic examples of enhancer grammar within Ciona and testing these examples through high-throughput, whole embryo reporter screens. I also introduce a novel, proof-of-concept package for performing flexible genomic searches of enhancer grammar with intuitive visualizations. Additionally, I share our current understanding of Ciona gastrulation and notogenesis from studying single-cell transcriptional expression profiles. Finally, I also discuss my contributions to Bioinformatics education.
    
\end{dissertationintroduction}
