\begin{dissertationintroduction}
    As the defining structure of all chordates, the notochord plays a crucial role in signaling and coordinating development during embryogenesis. In most vertebrates, the notochord ossified into the vertebrae of the spine. However, the notochord persists throughout the life of some invertebrate chordates, such as amphioxus. This thesis dissertation focuses on understanding gene regulation in the notochord of the marine urochordate, \textit{Ciona intestinalis} (\textit{Ciona}), during embryonic development from the perspective of the genomic sequence and the perspective of active transcripts within this key structure.

    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    \section{Notochord development in \textit{Ciona intestinalis}}
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    Chordates are animals belonging to the phylum Chordata, which includes vertebrates (subphylum Vertebrata), tunicates (subphylum Tunicata), and cephalochordates (subphylum Cephalochordata) \cite{holland2005}. The key defining characteristic of all chordates is the presence of a notochord during embryonic development \cite{stemple2004, holland2005, stemple2005, corallo2015, balmer2016, debree2018}. The notochord is a long, semi-rigid fibrous rod of mesodermal origin that provides structural support to the developing embryo along the anterior-posterior axis. The notochord also acts as a signaling center in the developing embryo, patterning structures such as the “neural tube” \cite{stemple2005, corallo2015, balmer2016}. A sheath of collagen proteins encases the notochord, allowing this flexible yet rigid structure to provide the basis for controlled mechanical support of Chordate organisms and protection for the neural tube \cite{corallo2015, stemple2004, stemple2005, balmer2016}.

    While some Chordates retain the notochord throughout life as their body’s primary axial support, in most vertebrates, the notochord becomes the nucleus pulposus of the intervertebral disc \cite{stemple2005, corallo2015, lawson2015, balmer2016}. Failure of vertebral notochord cells to transition to the nucleus pulposus is associated with chordomas. These slow-growing tumors form from notochord cell remnants within the spine or the base of the skull \cite{corallo2015, debree2018}. A complete understanding of notochord structure and function during early and late life stages is thus essential to better understand congenital neural tube and vertebral defects.

    As a close chordate relative to the vertebrates, the ascidian \textit{Ciona intestinalis Type A} or \textit{Ciona robusta} (\textit{Ciona}) stands as a longstanding model for studying organogenesis in a simple embryo \cite{dehal2002, delsuc2006, imai2006, satoh2014, satou2019, winkley2020}. For example, the \textit{Ciona} notochord consists of only 40 post-mitotic cells, and orthologs of many \textit{Ciona} notochord genes have known notochord expression in vertebrate embryos \cite{reeves2017, satoh2014, winkley2020}. Of the 40 notochord cells, 32 are grouped in the anterior of the body and compose the “primary” or “A-line” notochord. The remaining eight are located more posteriorly and form the “secondary” or “B-line” notochord \cite{satoh2014, winkley2020}. A-line and B-line refer to the conventional nomenclature denoting particular cell lineages in \textit{Ciona}. In the 4-cell \textit{Ciona} embryo, “A-lineage” and “B-lineage” cells are defined as the two cells on the vegetal side of the embryo, whereas the “a-lineage” and “b-lineage” cells are defined as the two cells on the animal side. The notochord thus forms from the vegetal A-line and B-line cells of the 4-cell \textit{Ciona} embryo \cite{satoh2014}.

    Within \textit{Ciona}, notochord precursor cells are defined as early as the eight-cell stage as the A4.1 and B4.1 blastomere pair in the developing anterior and posterior regions of the embryo, respectively \cite{corbo1997, satoh2014, yagi2004}. The A4.1 cells then divide to form the A5.1 and A5.2 blastomere pair at the onset of the 16-cell stage, which are precursors to the A-line notochord and the endoderm, nerve cord, trunk lateral cells, and muscle \cite{satoh2014}. On the other hand, the B4.1 cells divide to form the B5.1 and B5.2 blastomere pair and, through subsequent divisions from B5.1, divide into B6.1 and B6.2. Finally, the B6.1 blastomere descendant at the 32-cell stage will eventually develop into the B-line notochord and other mesenchymal and muscle cells \cite{corbo1997, satoh2014, yagi2004}. When gastrulation initiates at the 110-cell stage, the \textit{Ciona} embryo contains 16 primary and four secondary notochord precursor cells \cite{satoh2014}. Gastrulation is the stage at which the structure of the embryo changes from a single-layered blastula into a multiple-layered gastrula; thus, the notochord precursor cells coordinately invaginate as a monolayer over the primary gut, or archenteron \cite{rhee2005, winkley2020}. Following gastrulation is neurulation, the stage at which the embryonic neural plate develops and then forms the neural tube \cite{rhee2005, satoh2014}. At this stage, the notochord precursor cells in the \textit{Ciona} embryo divide for the last time to define the final set of notochord cells on the embryonic midline \cite{nakamura2012}. 

    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    \section{Elucidating the mechanisms regulating notogenesis}
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    The massive developmental transitions during embryogenesis require accurate gene regulation to maintain and balance the differentiation process. One component of this machinery is the interactions between cis-acting DNA elements-such as promoters and enhancers-and regulatory transcription factors. With the first example discovered in the early 1980s, enhancers are short regions of DNA that contain transcription factor binding sites (TFBSs) which proteins proteins can bind to regulate gene transcription \cite{khoury1983, kvon2021, levine2010}. Additionally, enhancers are typically located distally from the gene promoter and are approximately 100 bp to 1,000 bp in length \cite{khoury1983, levine2010}. Interestingly, the presence of a collection of TFBSs alone is insufficient in encoding functional activity of a particular target gene. For example, only specific arrangements of binding sites can activate transcription. The overarching rules governing the functional arrangement of TFBSs within enhancers is termed "enhancer grammar." Enhancer grammar is the interplay between the syntax-the order, orientation, and spacing of TFBSs-and the binding affinity of TFBSs to confer expression of a given enhancer sequence \cite{arnone1997, jindal2021}. Despite the importance of enhancers and their known association with developmental defects and disease, we still do not entirely understand how an enhancer’s sequence encodes particular functions. In Chapter 1, I will discuss the investigation into a notochord enhancer governed by Zic, ETS, FoxA, and Brachyury (Bra) transcription factor binding sites \cite{farley2016, song2022}. Zic and ETS are co-expressed in the developing notochord of \textit{Ciona} and in other vertebrates and are important for notochord specification \cite{dykes2018,matsumoto2007a}. The preceding study which discovered a putative notochord grammar relying on Zic and ETS found an interplay between the syntax and affinity of the binding sites present, such that the organization could compensate for the affinity and vice versa \cite{farley2016}. In Chapter 1, I will discuss an enhancer screen in which I search for evidence of the Zic and ETS notochord enhancer grammar across the \textit{Ciona} genome and test for functionality in the \textit{Ciona} notochord through a pilot screen of 90 genomic elements at the embryonic tailbud stage. From this screen, we were able to identify nine notochord enhancers, finding that enhancer grammar is critical within one of these elements. We also identify that some enhancers contain TFBSs for Zic, ETS, FoxA, and Bra, and translate that this set of binding sites may be an important signature for Brachyury enhancers across Chordates \cite{song2022}. 
    
    Beyond the universal quality of containing transcription factor motifs, enhancer sequences can vary significantly in the location, length, and type of transcription factor binding sites present. Additionally, these changes can be even more dramatic as you compare across species \cite{villar2015, ward2012, wong2020}. However, studies have suggested that even with low sequence conservation, the function of specific enhancers may be conserved across species and that this function may be partly due to combinatorial action of conserved transcription factors \cite{claussnitzer2014, wong2020}. This may be because a single transcription factor across its homologs in multiple species may have similar binding properties and thus recognize identical DNA sequences \cite{peter2011, wong2020}. In Chapter 2, I will continue the discussion of the notochord enhancer grammar studied in Chapter 1 but in greater detail and at a larger scale across the \textit{Ciona} genome. Within this study, we develop improvements over our initial search of \textit{Ciona} genomic regions containing Zic and ETS, such as allowing for greater flexibility of the Zic binding site within a sequence window. We find 4,344 genomic regions that harbor at least one Zic binding site and two ETS binding sites and test these regions in a massively-parallel reporter assay. In Chapter 2, I describe our preliminary results which suggest that only 9\% of the tested elements are functional enhancers. Further study of this enhancer library will likely identify novel notochord enhancers and help us better understand how Zic and ETS encode notochord development through particular grammatical constraints. 
    
    Within Chapter 1 and Chapter 2, I conduct high-throughput screens of genomic elements within developing whole embryos to better understand how enhancers encode notochord-specific expression patterns. Nonetheless, understanding the underlying processes driving development also requires understanding how genes are expressed, primarily how these gene expression profiles differ across cells \cite{peter2011}. For instance, all cells in a developing embryo contain the same set of genes. However, different cells express different sets of these genes, leading to differences in expression and, thus, molecular function \cite{arnone1997, peter2011}. Technological advances have enabled the cataloging of global gene expression profiles of single cells using single-cell RNA-sequencing (scRNA-seq), allowing scientists to define the heterogeneity within cell populations during embryonic development \cite{klein2015a, macosko2015, olsen2018}. This new paradigm has allowed developmental biologists to identify precisely when and in which cell types genes controlling cell fate decisions are expressed \cite{klein2019}. Despite the availability of large cell-type atlases generated via scRNA-seq and other omics technologies, there is still much to be learned about gene regulatory networks. \textit{Ciona} is a particularly suitable model for understanding the transcriptional changes necessary for proper development due to its genomic and morphological simplicity and historical significance as a model organism for embryological studies. In Chapter 3, I discuss an initiative to develop a high-resolution, single-cell atlas of a gastrulating \textit{Ciona} embryo to understand notogenesis and the formation of other early structures.  
    
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    \section{Training the next generation of bioinformaticians}
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    In recent years, genomics technologies have become more high-throughput and affordable to all research groups, resulting in a boom in data available for all biomedical research areas. However, this also results in a backlog of data to analyze for those that conducted the experiments. Despite never receiving a formal education in computation, many researchers are then faced with the arduous task of learning how to run bioinformatics pipelines \cite{barone2017, stephens2015}. While computational courses have started being integrated into the standard curriculum for undergraduate Biology majors, there remains a need to support graduate students and other scientists that did not experience this shift in training for the field. In Chapter 4, I will discuss the pedagogical philosophy that drove the in-person, hybrid, and virtual Bioinformatics courses I taught at the University of California, San Diego. 

    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
    \section{Conclusion}
    %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

    The massive developmental transitions during embryogenesis require accurate gene regulation to maintain and balance the differentiation process. In this dissertation, I present our approach to understanding regulation in the developing notochord by conducting high-throughput, whole embryo reporter screens to identify functional enhancers. I also present a novel, proof-of-concept package for performing flexible genomic searches of combinatorial arrangements of TFBSs. Additionally, I share our current understanding of \textit{Ciona} gastrulation and notogenesis from studying single-cell transcriptional expression profiles. Finally, I also discuss my contributions to Bioinformatics education.
    
\end{dissertationintroduction}
