# Decoding the genomic regulatory syntax driving notochord development

Within this folder, you can find all of the files used to compile my thesis dissertation in the form of `.tex` files. 

## Chapter Descriptions 

### Introduction

> **Primary File Name:** `1_introduction.tex`

As the defining structure of all chordates, the notochord plays a crucial role in signaling and coordinating development during embryogenesis. In most vertebrates, the notochord ossified into the vertebrae of the spine. However, the notochord persists throughout the life of some invertebrate chordates, such as amphioxus. This thesis dissertation focuses on understanding gene regulation in the notochord of the marine urochordate, *Ciona intestinalis* (*Ciona*), during embryonic development from the perspective of the genomic sequence and the perspective of active transcripts within this key structure.

Within this section, I delineate the various themes referenced in my thesis along with the necessary background information with the following subheadings:

* Notochord development in *Ciona intestinalis*
* Elucidating the mechanisms regulating notogenesis
* Training the next generation of bioinformaticians

### Chapter 1: Diverse logics encode notochord enhancers

> **Primary File Name:** `2_diverse-logics-encode-notochord-enhancers.tex`

This chapter provides a reformatted  reprint  of  the  material as it appeared in “Diverse logics encode notochord enhancers.”  Benjamin P. Song, Michelle F. Ragsac, Krissie Tellez, Granton A. Jindal, Jessica L. Grudzien, Sophia H. Le, Emma K. Farley. \textit{In Submission}, 2022.  

The contents of this chapter are now published and can be found in the following publication:

> **Song**, B.P., **Ragsac**, M.F., Tellez, K., Jindal, G.A., Grudzien, J.L., Le, S.H. and Farley, E.K., 2023. Diverse logics and grammar encode notochord enhancers. Cell reports, 42(2). https://www.cell.com/cell-reports/fulltext/S2211-1247(23)00063-3.

### Chapter 2: A proof-of-concept method to identify enhancers using constraints on binding site motifs

> **Primary File Name:** `3_proof-of-concept-grammar-searches.tex`

This chapter contains a proof-of-concept study that was started to follow up on the findings from Chapter 1. Within this work, we propose a proof-of-concept computational framework to search for tissue-specific enhancers within genomes using one's knowledge of TFBS motif signatures.

### Chapter 3: Understanding *Ciona intestinalis* gastrulation at single-cell resolution

> **Primary File Name:** `4_single-cell-ciona-gastrualtion.tex`

Since its early days as a choice model organism for critical biologists such as Laurent Chabry, Ed Conklin, and Thomas Hunt Morgan, *Ciona intestinalis* (*Ciona*) has been known for its simple embryos, rapid development, and ease of manipulation for embryological studies. Additionally, as a member of the subphylum Urochordata, *Ciona* represents the simplest and most primitive chordate body plans as our closest invertebrate relative. Several groups have already provided insight into *Ciona* embryogenesis by constructing partial gene regulatory networks or focusing on tissue-specific gene expression changes during particular developmental time points. However, there is still much we can delineate from studying cell fate determination pathways. 

Technological advances have enabled the cataloging of global gene expression profiles of single cells using single-cell RNA-sequencing (scRNA-seq), allowing scientists to define the heterogeneity within cell populations during embryonic development. This new paradigm has allowed developmental biologists, including those that study *Ciona*, to identify precisely when and in which cell types genes controlling cell fate decisions are expressed. Indeed, a previous study by Cao \textit{et al.} (2019) developed a single-cell transcriptional atlas for more than 90,000 cells spanning the onset of gastrulation through the swimming tadpole stage in *Ciona*. Their atlas spanned the 4.5 hours post fertilization (hpf) time point to the 18 hpf time point, demonstrating the feasibility of atlas-scale, whole embryo single-cell methods in *Ciona* and other marine tunicates. 

This chapter explores gastrulation in *Ciona* at a higher resolution than before using scRNA-seq, incorporating approximately 350,000 cells into a transcriptional atlas spanning just the 4.5 hpf, 5.5 hpf, and 6.5 hpf time points. 

### Chapter 4: Generating open educational resources for university-level bioinformatics courses

> **Primary File Name:** `5_bioinformatics-education.tex`

Rapid advances in next-generation sequencing (NGS) technologies have improved accessibility for experimentalists to generate genomic data at scale, but the barrier to entry to learning the computational skills necessary to analyze these datasets remains high. Despite computational courses being slowly integrated into the classical undergraduate Biology curricula, the breadth of scientific and technical knowledge needed to succeed in bioinformatics courses renders them inaccessible to individuals with incomplete foundations.

For many bioinformatics graduate programs, there can be an expectation for trainees to already have a baseline knowledge of programming and bioinformatics pipeline development. Inevitably, there is usually a proportion of admitted students that are non-computational. Not addressing this knowledge gap amongst non-computational scientists contributes to issues with student retention and morale within the program, especially for students of minoritized backgrounds. To directly address this need, I made it my mission in graduate school to develop inclusive teaching strategies in academically diverse classrooms to provide students with the skills necessary to confidently perform and understand bioinformatics analysis. Additionally, I advocated for and succeeded in making expectations of incoming bioinformatics graduate students clearer to improve the retention of trainees. As a consequence of the quarantine in response to the global SARS-CoV-2 pandemic, I taught in-person and fully online modalities. 

This chapter covers my reflections on teaching bioinformatics at the University of California, San Diego.

### Epilogue 

> **Primary File Name:** `6_epilogue.tex`

Finally, this chapter contains an overview of my doctoral work and final reflections.

## Repository Acknowledgements 

To compile this document, I relied heavily on the files in the following 
GitHub repository: https://github.com/stevecheckoway/ucsddissertation.
