# Decoding the genomic regulatory syntax driving notochord development

**Author:** Michelle Franc Ragsac <br>
**University:** University of California, San Diego <br>
**Department:** Bioinformatics & Systems Biology (https://bioinformatics.ucsd.edu/)

## Committee Members

* Professor Emma K. Farley, *Chair*
* Professor Theresa Gaasterland, *Co-Chair*
* Professor Vineet Bafna
* Professor Christopher Benner
* Professor Xin Sun

## Abstract

Embryonic development across all vertebrates begins upon the fertilization of an egg by a sperm cell to become a single-celled zygote. Embryogenesis continues with various stages of division to eventually make up an entire organism. The processes governing development are finely orchestrated and include many participants, such as genes involved in gene regulatory networks and non-coding regions of DNA, or enhancers, to regulate the expression of those genes. Defects or perturbations to this strictly regulated machinery can lead to various clinical conditions, such as congenital heart disease. Thus, deepening our understanding of embryogenesis may help us understand the mechanisms driving congenital abnormalities as well as the evolution of developmental pathways. One defining characteristic of all chordate embryos is the presence of a notochord during development. The notochord is a long, semi-rigid fibrous rod of mesodermal origin that provides structural support and serves as a signaling center to pattern the neighboring neural tube, paraxial mesoderm, and gut. A complete understanding of notochord structure and function during early and late life stages is thus essential to better understand congenital vertebral defects. For example, failure of vertebral notochord cells to transition to the nucleus pulposus, the cushioning between intervertebral discs of the spine, is associated with chordomas, slow-growing tumors formed from notochord cell remnants within the spine or the base of the skull. The ascidian *Ciona intestinalis Type A* (*Ciona*) is a marine organism that is evolutionarily similar to vertebrates. Through electroporation, *Ciona* is readily amenable to high-throughput, high-resolution functional studies of cis-regulatory elements like enhancers in their native, whole-embryo context. To identify key notochord enhancers, I analyzed the importance of enhancer grammar–the transcription factor order, orientation, spacing, and binding affinity–in modulating notochord-specific expression. Next, I highlight the potential of single-cell RNA-sequencing to study the gene regulato ry networks governing notogenesis and their relationship to congenital abnormalities. This body of work provides new insight into the regulatory processes governing notochord development, providing direction for future efforts to improve our understanding of notochord-based diseases across chordates. Finally, I highlight Open Educational Resources (OERs) I developed for Bioinformatics education, emphasizing accessibility and inclusion.

## eScholarship Link for Dissertation

My dissertation can be found online on eScholarship under identifier `1471r9g0`: https://escholarship.org/uc/item/1471r9g0. 

On that page, it is possible to download the official PDF document, but I have also included it within this repository! 

### Citation Information

**Ragsac**, M. F. (2022). Decoding the genomic regulatory syntax driving notochord development. *UC San Diego*. ProQuest ID: `Ragsac_ucsd_0033D_22000`. Merritt ID: `ark:/13030/m5z39b30`. Retrieved from https://escholarship.org/uc/item/1471r9g0

## Repository Acknowledgements 

My dissertation document references the template that can be found at the following GitHub repository: https://github.com/stevecheckoway/ucsddissertation. 
